src/connectors/stripe/index.ts

```diff
@@ -361,11 +363,42 @@ class StripeService {
     }
   }
 
+  /**
+   * Get delivery failed events in last 7 days.
+   */
   getDeliveryFailedEvents = async () => {
     try {
-      const events = await this.stripeAPI.events.list({
-        delivery_success: false,
-      })
+      let cursor
+      let fetch = true
+      let hasMore = true
+
+      const now = Date.now()
+      const week = DAY * 7
+      const events: Array<Record<string, any>> = []
+
+      while (hasMore && fetch) {
+        // fetch events from stripe
+        const batch: Record<string, any> = await this.stripeAPI.events.list({
+          delivery_success: false,
+          limit: 10,
```

Comment:
To reduce the API requests, should we increase the `limit` to the upper (100)?

Reply:
👌🏻 Resize to 50.